packages/react/src/components/accordion/accordion-container.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState, ReactElement, createRef, useRef } from 'react';
+import { AccordionProps, AccordionContainerProps } from './accordion-types';
+import { Accordion } from './accordion';
+import { StyledAccordionGroup } from './accordion-styles';
+
+export const AccordionContainer: React.FC<AccordionContainerProps> = ({
+    id, mode = 'single', children,
+}) => {
+    const generateId = (childProps: AccordionProps, index: number): string => childProps.id || `${id}-${index}`;
+    const isAccordion = (child: React.ReactNode): child is ReactElement<AccordionProps> => (
+        React.isValidElement<AccordionProps>(child) && child.type === Accordion);
+    const focusedAccordionRef = useRef<number | null>(null);
+
+    const [expandedItemIds, setExpandedItemIds] = useState<string[]>(() => (
+        React.Children.toArray(children)
+            .filter(isAccordion)
+            .map((child: ReactElement<AccordionProps>, index) => {
+                const childProps = child.props as AccordionProps;
+                return childProps.isExpanded ? generateId(childProps, index) : null;
+            })
+            .filter((expandedId) => expandedId !== null) as string[]
+    ));
+
+    const handleToggle = (itemId: string): void => {
+        if (mode === 'single') {
+            setExpandedItemIds((prevIds) => (prevIds.includes(itemId) ? [] : [itemId]));
+        } else if (mode === 'multi') {
+            setExpandedItemIds((prevIds) => {
+                if (prevIds.includes(itemId)) {
+                    return prevIds.filter((expandedId) => expandedId !== itemId);
+                }
+                return [...prevIds, itemId];
+            });
+        }
+    };
+
+    const childrenArray: ReactElement<AccordionProps>[] = React.Children.toArray(children)
+        .filter(isAccordion)
+        .map((child: ReactElement<AccordionProps>, index) => {
+            const buttonRef = createRef<HTMLButtonElement>();
+            const childProps = child.props as AccordionProps;
```

Comment:
This is unnecessary and redundant, also same in map.

- child.props is already resolved as AccordionProps because child is ReactElement<AccordionProps>.
- child is resolved as `ReactElement<AccordionProps>` because of the type guard isAccordion

You would still have all the typing like this.

```
React.Children.toArray(children)
  .filter(isAccordion)
  .map(child, index) => {
     child.props
  }
```